components/engine_rocks/src/flush_listener.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    fmt::Debug,
+    sync::{Arc, Mutex},
+};
+
+use engine_traits::util::{
+    increase_memtable_version, max_synced_sequence_number, MemtableEventNotifier,
+};
+use parking_lot_core::SpinWait;
+use rocksdb::{EventListener, FlushJobInfo, MemTableInfo};
+use tikv_util::debug;
+
+use crate::RocksEngine;
+
+#[derive(Clone)]
+pub struct FlushListener {
+    notifier: Arc<Mutex<Box<dyn MemtableEventNotifier>>>,
+    engine: Arc<Mutex<Option<RocksEngine>>>,
```

Comment:
I think the locks can be removed in some way, you only modify them once.

Reply:
Do you mean using `once_cell`?

Reply:
Create the listener after engine is created.

Reply:
Forget it, the ordering is tricky.